service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -43,4 +43,23 @@
      * A list of service URIs.
      */
     public abstract List<String> uris();
+
+    // hides implementation details
+    public static Builder builder() {
+        return ImmutableServiceConfiguration.builder();
+    }
+
+    // hides implementation details
+    public interface Builder {
```

Comment:
What exactly do you mean by "implementation details"?

On Tue, Apr 19, 2016 at 11:17 AM Qinfeng Chen <notifications@github.com>
wrote:

> In
> service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java
> <https://github.com/palantir/http-remoting/pull/106#discussion_r60282799>:
>
> > @@ -43,4 +43,23 @@
> >       * A list of service URIs.
> >       */
> >      public abstract List<String> uris();
> > +
> > +    // hides implementation details
> > +    public static Builder builder() {
> > +        return ImmutableServiceConfiguration.builder();
> > +    }
> > +
> > +    // hides implementation details
> > +    public interface Builder {
>
> Would love to collect some best practices on immutables and add them in
> baseline. I created a quip doc here, please feel free to update it.
> https://palantir.quip.com/dDUfABPSL7eP
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/palantir/http-remoting/pull/106/files/511164691ff08f26a4da8c6c8ee8e99f7521bccf#r60282799>
>